src/predicate/isDefinedArray.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Check if all items in the array are defined.
+ *
+ * @param array The array to check.
+ * @returns `true` if all items in the array are defined, `false` otherwise.
+ *
+ * @example
+ * ```ts
+ * isDefinedArray([1, 2, 3]); // true
+ * isDefinedArray([undefined, 1, 2, 3]); // false
+ * isDefinedArray([null, 1, 2, 3]); // false
+ * isDefinedArray([undefined, null, 1, 2, 3]); // false
+ * isDefinedArray([1, 2, 3, undefined]); // false
+ * isDefinedArray([1, 2, 3, null]); // false
+ * isDefinedArray([1, 2, 3, undefined, null]); // false
+ * ```
+ */
+export function isDefinedArray<T>(array: Array<T | undefined | null>): array is T[] {
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] === undefined || array[i] === null) {
+      return false;
+    }
+  }
+
+  return true;
```

Comment:
```suggestion
 return array.every(el=> el !== null && el !== undefined);
```
I think this makes code cleaner.